packages/features/auth/lib/next-auth-options.ts

```diff
@@ -922,6 +925,48 @@ export const AUTH_OPTIONS: AuthOptions = {
       return baseUrl;
     },
   },
+  events: {
+    async signIn({ user, isNewUser }) {
+      /* only run this code if:
+         - it's a hosted cal account
+         - DUB_API_KEY is configured
+         - it's a new user
+      */
+      const hostedCal = Boolean(HOSTED_CAL_FEATURES);
+      if (hostedCal && process.env.DUB_API_KEY && isNewUser) {
+        const dclid = cookies().get("dclid")?.value;
+        // here we use waitUntil – meaning this code will run async to not block the main thread
+        waitUntil(
+          Promise.allSettled([
+            // create a new short link for the user
+            // this will be used in the PoweredBy button on the Booking page
+            dub.links.create({
+              domain: "go.cal.com",
+              key: user.username || undefined,
```

Comment:
I wonder how we can get this username stuff working with orgs. @hariombalhara  

For ref @steven-tey usernames arent unique anymore 

e.g. - cal.com/sean & i.cal.com/sean are different users.

Reply:
We have similar handling at https://github.com/calcom/cal.com/blob/de46b1f1221f42e5a8399a00b698494406c15295/packages/features/auth/lib/next-auth-options.ts#L645 Maybe that can help.

Reply:
@sean-brydon maybe we can limit this to just `cal.com/{username}`? 

Also cc: @PeerRich who mentioned that we should use `refer.cal.com/{username}` instead of `go.cal.com/r/{username}` for the short links.

Reply:
I think this is a fair shout - i think a lot of people in orgs will probably have the "powered by" bit disabled. From what i can see thats the main "referal" point. 

Nice job so far @steven-tey <3